src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -6378,8 +6378,8 @@ private static String replace(final String text, String searchString, final Stri
          }
          final int replLength = searchString.length();
          int increase = replacement.length() - replLength;
-         increase = increase < 0 ? 0 : increase;
-         increase *= max < 0 ? 16 : max > 64 ? 64 : max;
+         increase = Math.max(increase, 0);
+         increase *= Math.min(max, 0);
```

Comment:
How can this be correct when the bounds above are 16 and 64?

Reply:
That is a very good spot, not sure what happened here. I've fixed it now. Interesting that no tests picked it up.